skyportal/handlers/api/internal/token.py

```diff
@@ -135,6 +137,118 @@ def get(self, token_id=None):
+        tags:
+          - telescopes
+        parameters:
+          - in: path
+            name: token_id
+            required: true
+            schema:
+              type: integer
+        requestBody:
+          content:
+            application/json:
+              schema: Token
+        responses:
+          200:
+            content:
+              application/json:
+                schema: Success
+          400:
+            content:
+              application/json:
+                schema: Error
+        """
+
+        with self.Session() as session:
+            try:
+                token = session.scalars(
+                    Token.select(session.user_or_token, mode="update").where(
+                        Token.id == token_id
+                    )
+                ).first()
+                if token is None:
+                    return self.error(
+                        'Either the specified token does not exist, '
+                        'or the user does not have the necessary '
+                        'permissions to delete it.'
+                    )
+
+                data = self.get_json()
+                data['id'] = token_id
+
+                if 'user_id' in data:
+                    user_id = data['user_id']
+                    user = session.scalars(
+                        User.select(session.user_or_token).where(User.id == user_id)
+                    ).first()
+                else:
+                    user = self.associated_user_object
+                    user_id = user.id
+
+                schema = Token.__schema__()
+                try:
+                    schema.load(data, partial=True)
+                except ValidationError as e:
+                    return self.error(
+                        'Invalid/missing parameters: ' f'{e.normalized_messages()}'
+                    )
+                if 'name' in data:
+                    token.name = data['name']
+
+                if 'acls' in data:
+                    token_acls = set(data['acls'])
+                    if not all([acl_id in user.permissions for acl_id in token_acls]):
+                        return self.error(
+                            "User has attempted to grant token ACLs they do not have "
+                            "access to. Please try again."
+                        )
+
+                    new_acl_ids = list(set(token_acls))
+                    if not all(
+                        [
+                            session.scalars(
+                                ACL.select(session.user_or_token).where(
+                                    ACL.id == acl_id
+                                )
+                            ).first()
+                            is not None
+                            for acl_id in new_acl_ids
+                        ]
+                    ):
+                        return self.error(
+                            "Improperly formatted parameter aclIds; must be an array of strings."
```

Comment:
It seems like you are not just testing that they are strings, you are also checking they are legit ACLs. Maybe the error message should reflect that.

Reply:
Good point